internal/compiler-bridge/src/main/scala/xsbt/ExtractUsedNames.scala

```diff
@@ -177,6 +240,30 @@ class ExtractUsedNames[GlobalType <: CallbackGlobal](val global: GlobalType)
       }
     }
 
+    val addSymbolPosition: (NamePositionMap, Symbol, Tree, Role) => Unit = {
+      (namePositions: NamePositionMap, symbol: Symbol, tree: Tree, role: Role) =>
+        if (tree.pos != NoPosition) {
+          // If more than one reference is included in one position, only the first one is registered.
+          val registeredReference = namePositions.containsKey(tree.pos) && role == Role.REFERENCE
+          val ignoreReference = {
+            val tpe = symbol.tpe
+            role == Role.REFERENCE && (definitions.isByNameParamType(tpe) || definitions
+              .isRepeatedParamType(tpe))
+          }
+          if (!registeredReference && !ignoreReference && !symbol.hasPackageFlag && !ignoredSymbol(
+                symbol) && !isEmptyName(symbol.name)) {
+            val semanticName = symbol.toSemanticName
+            // Local names are not collected.
+            if (!semanticName.isEmpty) {
```

Comment:
Does this mean goto definition won't work for local variables?

Reply:
No, I think the language server that uses this index should analysis the local variables separately and return it to the client.
Since it is enough to analyze only one file, it should be able to execute at low cost.
Considering the enlargement of index size, and performance, I omitted it.